Mathlib/Topology/Algebra/ProperAction.lean

```diff
@@ -0,0 +1,277 @@
+/-
+Copyright (c) 2024 Anatole Dedeker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anatole Dedeker, Etienne Marion, Florestan Martin-Baillon, Vincent Guirardel
+-/
+import Mathlib.Topology.Algebra.Group.Basic
+import Mathlib.Topology.Algebra.MulAction
+import Mathlib.Topology.Maps.Proper.Basic
+import Mathlib.Topology.Sequences
+
+/-!
+# Proper group action
+
+In this file we define proper action of a group on a topological space, and we prove that in this
+case the quotient space is T2. We also give equivalent definitions of proper action using
+ultrafilters and show the transfer of proper action to a closed subgroup.
+We give sufficent conditions on the topological space such that the action is properly discontinuous
+(see `ProperlyDiscontinuousSMul`) if and only if it is continuous in
+the first variable (see `ContinuousConstSMul`) and proper in the sense defined here.
+
+## Main definitions
+
+* `ProperSMul` : a group `G` acts properly on a topological space `X`
+  if the map `(g, x) ↦ (g • x, x)` is proper, in the sense of `IsProperMap`.
+
+## Main statements
+
+* `t2Space_quotient_mulAction_of_properSMul`: If a group `G` acts properly
+  on a topological space `X`, then the quotient space is Hausdorff (T2).
+* `t2Space_of_properSMul_of_t2Group`: If a T2 group acts properly on a topological space,
+  then this topological space is T2.
+* `ProperlyDiscontinuousSMul_iff_ProperSMul`: If a discrete group acts on a T2 space `X` such that
+  `X × X` is compactly generated, then the action is properly discontinuous if and only if it is
+  continuous in the second variable and proper. This in particular true if `X` is locally compact
+  or first-countable.
+
+## Implementation notes
+
+Concerning `ProperlyDiscontinuousSMul_iff_ProperSMul`, this result should be the only one needed
+to link properly discontinuous and proper actions, however the notion of compactly generated space
+is not yet in Mathlib so it is used as an intermediate result to prove
```

Comment:
Can you update the wording to reflect the new status of compactly generated spaces. Note: you do *not* need to try and add the relevant results.